app/bulk_resource/views.py

```diff
@@ -173,22 +188,120 @@ def review_options():
                            csv_container=csv_container,
                            form=form)
 
+@bulk_resource.route('/get-required-option-descriptor', methods=['GET', 'POST'])
```

Comment:
let's add some comments in this function and the next since they're fairly long and complex